### PR TITLE
fix: Plugin testing sharp edges

### PR DESCRIPTION
**Pull request by martijneken**
Various small fixes

- error when header key is omitted
- don't benchmark stream teardown

### CODE REVIEW

File: plugins/test/dynamic_test.cc [hunk 1]
```diff
@@ -320,6 +320,7 @@ void DynamicTest::BenchHttpHandlers(benchmark::State& state) {
       benchmark::DoNotOptimize(res);
       BM_RETURN_IF_FAILED(handle);
     }
+    state.PauseTiming();
```
Comment (leonm1) [inline plugins/test/dynamic_test.cc:323]: The benchmarking framework's documentation says this should be avoided in loops: https://github.com/google/benchmark/blob/25e5c52a112a56acdc37cace73025f7327f03be7/include/benchmark/benchmark.h#L805

Does this make a measurable difference in the outcome?

**Review by leonm1 — APPROVED** [review]

File: plugins/test/dynamic_test.cc (hunk 1 above)
Comment (martijneken) [reply plugins/test/dynamic_test.cc:323]: Ah, interesting guidance, I hadn't seen that. The problem is that right now we call PauseTiming/ResumeTiming above, which excludes TestHttpContext construction cost. But then the TestHttpContext teardown cost is included here, so we are including half of BenchStreamLifecycle in this benchmark.

I would be OK with one model or the other, but not halfway, hence this quick fix.

Long term I think I'd prefer to get rid of these calls (thus including the TestHttpContext lifecycle costs). To do that, we'd also have to stop handing ownership of BODY chunks to SendRequestBody. @mpwarres FYI

Comment (martijneken) [reply plugins/test/dynamic_test.cc:323]: Heh, but at the same time this seems to cause asan/tsan failures, presumably on the last iteration... `StopTimer: Check 'running_' failed.`

Comment (martijneken) [reply plugins/test/dynamic_test.cc]: Fixed a different way (std::optional stack variable outside the benchmark loop). Left a comment about possible future work.

**Pull request merged**